R/FilteredDataset.R

```diff
@@ -212,64 +212,87 @@ FilteredDataset <- R6::R6Class( # nolint
         id = id,
         include_css_files("filter-panel"),
         include_js_files(pattern = "icons"),
-        tags$div(
-          id = ns("whole_ui"), # to hide it entirely
-          fluidRow(
-            style = "padding: 0px 15px 0px 15px;",
+        bslib::accordion(
+          id = ns("dataset_filter_accordian"),
+          class = "teal-slice-dataset-filter",
+          bslib::accordion_panel(
+            dataname,
+            icon = icon("fas fa-table"),
             tags$div(
-              style = "display: flex; align-items: center; justify-content: space-between;",
-              tags$div(
-                style = "display: flex;",
-                tags$span(dataname, class = "filter_panel_dataname"),
+              id = ns("whole_ui"), # to hide it entirely
+              div(
+                id = ns("filter_util_icons"),
+                style = "display: flex; flex-direction: row-reverse; align-items: center; width: 3.5rem; margin-left: -3.5rem; z-index: 99;",
                 if (allow_add) {
                   tags$a(
                     class = "filter-icon add-filter",
+                    style = "font-weight: 100; font-size: 1.3rem; color: black; padding: 0.2rem;",
                     tags$i(
                       id = ns("add_filter_icon"),
                       class = "fa fa-plus",
                       title = "fold/expand transform panel",
                       onclick = sprintf(
-                        "togglePanelItems(this, '%s', 'fa-plus', 'fa-minus');",
-                        ns("add_panel")
+                        "togglePanelItems(this, '%s', 'fa-plus', 'fa-minus');
+                        if ($(this).hasClass('fa-minus')) {
+                          $('#%s .accordion-button.collapsed').click();
+                        }",
+                        ns("add_panel"),
+                        ns("dataset_filter_accordian")
```

Comment:
@gogonzo I'm using the accordion open/close in cases where I can trigger using R. Here, it is not important for this icon to create shiny binding so simply clicking the accordion button seemed like an acceptable solution.